src/mlpack/methods/decision_tree/decision_tree.hpp

```diff
@@ -123,6 +124,18 @@ class DecisionTree :
                    typename std::remove_reference<WeightsType>::type>::value>*
                     = 0);
 
+  template<typename MatType, typename LabelsType, typename WeightsType>
+  DecisionTree(const DecisionTree& other,
+               MatType data,
+               const data::DatasetInfo& datasetInfo,
+               LabelsType labels,
+               const size_t numClasses,
+               WeightsType weights,
+               const size_t minimumLeafSize = 10,
+               const double minimumGainSplit = 1e-7,
+               const std::enable_if_t<arma::is_arma_type<
+                       typename std::remove_reference<WeightsType>::type>::value>*
+               = 0);
```

Comment:
Can we add some documentation for this function?  Also, the wrapped line for the SFINAE bit only needs to be indented 4 characters, which should let the `= 0` fit on the line above and clean it up a bit. :)